src/js/components/product-select/ProductSelect.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+
+import { debounceProductsFetch } from 'utils/option-utils';
+import renderHandlingIcons from 'utils/product-handling-icons';
+import Select from 'utils/Select';
+
+const Option = option => (
+  <strong style={{ color: option.color || 'black' }} className="d-flex align-items-center">
+    {option.label}
+  &nbsp;
+    {renderHandlingIcons(option.handlingIcons)}
+  </strong>);
+
+const SelectedValue = option => (
+  <span className="d-flex align-items-center">
+    <span className="text-truncate">{option.label}</span>
+    &nbsp;{renderHandlingIcons(option ? option.handlingIcons : [])}
+  </span>
+);
+
+const ProductSelect = ({
+  onExactProductSelected, locationId, fieldRef, ...props
+}) => {
+  const selectRef = useRef(null);
+  const [isExactMatch, setIsExactMatch] = useState(false);
+  const [loadedOptions, setLoadedOptions] = useState([]);
+  const {
+    debounceTime, minSearchLength,
+  } = useSelector(state => ({
+    debounceTime: state.session.searchConfig.debounceTime,
+    minSearchLength: state.session.searchConfig.minSearchLength,
+  }));
+
+  const debouncedProductsFetch = debounceProductsFetch(
+    debounceTime,
+    minSearchLength,
+    locationId,
+  );
+
+  const onEnterPress = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsExactMatch(true);
+  };
+
+  useEffect(() => {
+    if (isExactMatch && loadedOptions.length > 0) {
+      const exactMatchProduct = loadedOptions.find(product => product.exactMatch);
+
+      if (onExactProductSelected) {
+        onExactProductSelected({ isExactMatch, product: exactMatchProduct });
```

Comment:
what is that `isExactMatch` provided here for? From what I can see in `fields` config, we are only destructuring `product`

Reply:
I thought maybe this value might be useful in some way but I guess I didn't use it anywhere, so lets stick with the YAGNI principle  :D  and I'll remove that